File: backend/src/api/events/events.controller.ts
import { Body, Controller, HttpCode, Post } from '@nestjs/common'
import { EventsService } from '../../application/events/events.service'
import { CreateEventDto } from '../../contracts/dto/event.dto'

/* Public analytics ingress — no auth, anonymous by design. The SPA's track()
   util fires here from call sites (save/share/directions/pin/filter/submission). */
@Controller('events')
export class EventsController {
  constructor(private readonly events: EventsService) {}

  @Post()
  @HttpCode(202)
  track(@Body() body: CreateEventDto): Promise<{ ok: true }> {
    return this.events.track(body)
  }
}

File: backend/src/main/java/ee/gemspot/api/web/EventsController.java
package ee.gemspot.api.web;

import ee.gemspot.api.dto.CreateEventDto;
import ee.gemspot.api.dto.OkDto;
import ee.gemspot.api.service.EventsService;
import jakarta.validation.Valid;
import org.springframework.http.HttpStatus;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.ResponseStatus;
import org.springframework.web.bind.annotation.RestController;

/** Public analytics ingress — anonymous by design. 202 Accepted. */
@RestController
@RequestMapping("/events")
public class EventsController {

    private final EventsService events;

    public EventsController(EventsService events) {
        this.events = events;
    }

    @PostMapping
    @ResponseStatus(HttpStatus.ACCEPTED)
    public OkDto track(@Valid @RequestBody CreateEventDto body) {
        events.track(body);
        return new OkDto(true);
    }
}
